### PR TITLE
Xeno slashes no longer attempt to slash destroyed (delimbed) limbs

### PR DESCRIPTION
**Pull request by ma44**
## About The Pull Request
Xenomorphs attacking a limb that was destroyed would play the slash animation and sound but do no damage as the limb is technically completely removed, now if a xeno requests a limb to slash, it will always be a limb that isn't destroyed

## Why It's Good For The Game
Stops occasions where xenos don't do any slashing damage because their targeting a delimbed limb or the RNG limb targeting picked a limb that isn't possible to do slash damage to

## Changelog
:cl: ma44
fix: Xenomorphs will no longer attempt to slash destroyed (delimbed) limbs
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/attack_alien.dm [hunk 1]
```diff
@@ -141,17 +141,17 @@
...
 		affecting = get_limb(set_location)
 	else if(X.stealth_router(HANDLE_STEALTH_CHECK) && X.stealth_router(HANDLE_SNEAK_ATTACK_CHECK)) //We always get the limb we're aiming for if we're sneak attacking
 		affecting = get_limb(X.zone_selected)
 	else
 		affecting = get_limb(ran_zone(X.zone_selected, 70))
-	if(!affecting || (random_location && !set_location)) //No organ, just get a random one
+	if(!affecting || (random_location && !set_location) || (ignore_destroyed && !affecting.is_usable())) //No organ or it's destroyed, just get a random one
 		affecting = get_limb(ran_zone(null, 0))
-	if(no_head && affecting == get_limb("head"))
+	if(no_head && affecting == get_limb("head") || (ignore_destroyed && !affecting.is_usable()))
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/attack_alien.dm]: What is this line for?

Comment (ma44) [reply code/modules/mob/living/carbon/xenomorph/attack_alien.dm]: is_usable() returns TRUE if the limb isn't destroyed, mutated or necrotized. if ignore_destroyed is TRUE then the proc won't try to return a unusable limb

Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/attack_alien.dm]:
```suggestion
	if(!affecting || (no_head && affecting == get_limb("head")) || (ignore_destroyed && !affecting.is_usable()))
```
And delete the next check `if(!affecting)` check.

File: code/modules/mob/living/carbon/xenomorph/attack_alien.dm [hunk 2]
```diff
@@ -141,17 +141,17 @@
...
+/mob/living/carbon/get_xeno_slash_zone(mob/living/carbon/xenomorph/X, set_location = FALSE, random_location = FALSE, no_head = FALSE, ignore_destroyed = TRUE)
 	var/datum/limb/affecting
 	if(set_location)
 		affecting = get_limb(set_location)
 	else if(X.stealth_router(HANDLE_STEALTH_CHECK) && X.stealth_router(HANDLE_SNEAK_ATTACK_CHECK)) //We always get the limb we're aiming for if we're sneak attacking
 		affecting = get_limb(X.zone_selected)
 	else
 		affecting = get_limb(ran_zone(X.zone_selected, 70))
-	if(!affecting || (random_location && !set_location)) //No organ, just get a random one
+	if(!affecting || (random_location && !set_location) || (ignore_destroyed && !affecting.is_usable())) //No organ or it's destroyed, just get a random one
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/attack_alien.dm]:
```suggestion
	if(random_location && !set_location) //No organ, just get a random one
```
Let's keep things clean and simple, without duplicate checks.

File: code/modules/mob/living/carbon/xenomorph/attack_alien.dm [hunk 3]
```diff
@@ -141,17 +141,17 @@
...
 		affecting = get_limb(ran_zone(null, 0))
-	if(no_head && affecting == get_limb("head"))
+	if(!affecting || (no_head && affecting == get_limb("head")) || (ignore_destroyed && !affecting.is_usable()))
 		affecting = get_limb("chest")
 	if(!affecting) //Still nothing??
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/attack_alien.dm]: Remove this now.

**Pull request merged**